### ISSUE TITLE
MediaStreamTrack.stop() should be async

### ISSUE THREAD

**Issue by sonntag-philipp** [w3c/mediacapture-main#817]
Hello there,

I'm currently working with the mediacapture api in different browsers. While working on the MediaStreamTracks, I've noticed that the stop method doesn't provide a way to get notified when the track is actually stopped by the underlying user agent. 

This can cause issues in platforms, where a camera can only be opened once by the browser api.

My proposal for a solution would be, to make the .stop() method async by returning a promise.
The promise should be resolved, when the user agent made sure, that the media track is stopped.

**Comment by fippo** [comment]
does that require an API change or should the user agent ensure pending closes are done before attempting to open a device?

**Comment by sonntag-philipp** [comment]
Yes, I think it would require an api change as the method signature of `MediaStreamTrack.stop()` would change to return a promise.
Anyway, this wouldn't affect existing uses of the method. They could just ignore the return value.

**Comment by guidou** [comment]
I agree that returning a promise would be useful and allow more flexibility for implementation.
